opencog/cogserver/server/Request.cc

```diff
@@ -45,14 +47,41 @@ Request::~Request()
         _console->OnRequestComplete();
         _console->put();  // dec use count we are done with it.
     }
+
+    // Normally the thread object will be cleared when it is done.
+    if (_thread)
+    {
+        if (_thread->joinable())
+            _thread->join();
+        delete _thread;
+    }
+}
+
+void Request::executeAsynch(void)
+{
+    // Execute in a new thread.
+    auto execute_wrapper = [&](void) 
+    {
+        try{
+            execute();
+        } catch (...) {
+            logger().debug("Request::executeAsynch caught exception!");
+        }        
+        
+        // Always tell the CogServer so this request is deleted
+        // even in the event of an exception.
+        _cogserver.asynchRequestDone(this);
+    };
+    
+    _thread = new std::thread(execute_wrapper);
```

Comment:
There are no asynchronously running requests by default. All requests get run synchronously unless they create their own threads.

I thought I'd add support for asynchronous requests, turned off by default, so that one could implement a request that executed in it's own thread to mirror what happens when you netcat with a shell command, these get their own threads.

Requests were not handled this way before.